### ISSUE TITLE
MPS cumsum failure for 5D tensor or above

### ISSUE THREAD

**Issue by manuelcandales** [pytorch/pytorch#154881]
### 🐛 Describe the bug

MPS cumsum errors out for 5D tensor or above, when the difference between `x.dim()` and `dim` is more than 4.
For example, the following cases fail:
```python
x = torch.ones([1, 2, 3, 4, 5], device="mps")
torch.ops.aten.cumsum(x, dim=0)
```
```python
x = torch.ones([1, 2, 3, 4, 5, 6], device="mps")
torch.ops.aten.cumsum(x, dim=1)
```

while the following cases succeed:
```python
x = torch.ones([1, 2, 3, 4, 5], device="mps")
torch.ops.aten.cumsum(x, dim=1)
```
```python
x = torch.ones([1, 2, 3, 4, 5, 6], device="mps")
torch.ops.aten.cumsum(x, dim=2)
```

Error message:
```
/AppleInternal/Library/BuildRoots/01adf19d-fba1-11ef-a947-f2a857e00a32/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSNDArray/Kernels/MPSNDArrayScan.mm:352: failed assertion `(null)" Axis = 4. This class only supports axis = 0, 1, 2, 3
'
zsh: abort
```

### Versions

torch==2.7.0

cc @ezyang @gchanan @zou3519 @kadeng @msaroufim @kulinseth @albanD @malfet @DenisVieriu97 @jhavukainen

**Comment by jhavukainen** [comment]
Thanks for filing the issue. Started off by adding an internal issue to follow up on MPSGraph side since this behavior is not documented on the cumulativeSumWithOp:axis:name op documentation and it can also be reproduced when directly calling MPSGraph with the arguments you provided.

**Comment by malfet** [comment]
Hi pri to raise structured error rather than crash

**Comment by jhavukainen** [comment]
Assigning my self to add an error if the difference between ndims() and axis of reduction > 4.
